src/viam/app/app_client.py

```diff
@@ -0,0 +1,784 @@
+        request = DeleteLocationRequest(location_id=location_id)
+        await self._location_client.DeleteLocation(request, metadata=self._metadata)
+
+    async def list_locations(self, organization_id: str) -> List[Location]:
+        """Get a list of all locations under the specified organization.
+
+        Args:
+            organization_id (str): ID of the organization to retrieve the locations from.
+
+        Raises:
+            GRPCError: If an invalid organization ID is passed.
+
+        Returns:
+            List[viam.proto.app.Location]: The list of locations.
+        """
+        request = ListLocationsRequest(organization_id=organization_id)
+        response: ListLocationsResponse = await self._location_client.ListLocations(request, metadata=self._metadata)
+        return response.locations
+
+    async def share_location(self, organization_id: str, location_id: Optional[str] = None) -> None:
+        """Share a location with a specific organization.
+
+        Args:
+            organization_id (str): ID of the organization to share the location with.
+            location_id (Optional[str]): ID of the location to be shared. Defaults to the current authorized location.
+
+        Raises:
+            GRPCError: Permission denied. A location can only be shared by users who are owners of both organizations involved. An
+                `AppClient` cannot be an authorized owner of two organizations. Permission will always be denied when calling this method. A
+                GRPCError is also thrown when attempting to share a location with an organization it is already shared with.
+        """
+        location_id = location_id if location_id else self._location_id
+        request = ShareLocationRequest(location_id=location_id, organization_id=organization_id)
+        await self._location_client.ShareLocation(request, metadata=self._metadata)
+
+    async def unshare_location(self, organization_id: str, location_id: Optional[str] = None) -> None:
+        """Unshare a location with a specific organization.
+
+        Args:
+            organization_id (str): ID of the organization to unshare the location with.
+            location_id (Optional[str]): ID of the location to be unshared. Defaults to the current authorized location.
+
+        Raises:
+            GRPCError: Permission denied. A location can only be unshared by users who are owners of both organizations involved. An
+                `AppClient` cannot be an authorized owner of two organizations. Permission will always be denied when calling this method. A
+                GRPCError is also thrown when attempting to unshare a location with an organization it is not already shared with.
+        """
+        location_id = location_id if location_id else self._location_id
+        request = UnshareLocationRequest(location_id=location_id, organization_id=organization_id)
+        await self._location_client.UnshareLocation(request, metadata=self._metadata)
```

Comment:
I think these implementations should be removed for now since we don't plan on supporting it right now - we can always add them in a pinch if we need to